iocage_lib/ioc_start.py

```diff
@@ -1312,19 +1311,27 @@ def __check_rtsold__(self):
             stdout=su.PIPE
         )
 
-    def get_default_gateway(self):
-        # e.g response - ('192.168.122.1', 'lagg0')
-        try:
-            return netifaces.gateways()["default"][netifaces.AF_INET]
-        except KeyError:
+    def get_default_interface(self):
+        if netifaces.AF_INET in self.host_gateways['default']:
+            return self.host_gateways['default'][netifaces.AF_INET][1]
+        elif netifaces.AF_INET6 in self.host_gateways['default']:
+            return self.host_gateways['default'][netifaces.AF_INET6][1]
+        else:
             iocage_lib.ioc_common.logit(
                 {
                     'level': 'EXCEPTION',
-                    'message': 'No default gateway interface found'
+                    'message': 'No default interface found'
                 },
                 _callback=self.callback,
-                silent=self.silent
-            )
+                silent=self.silent)
+
+    def get_default_gateway(self, address_family=netifaces.AF_INET):
+        if address_family in self.host_gateways['default']:
+            return re.sub(
+                '%.*$', '%epair0b',
```

Comment:
I am sorry, i do not follow this change, could you please let me know what purpose this serves ?

Reply:
Do you mean the `re.sub`?

As you can see in the `netstat` output down there `defaultrouter6 ` is a link local address with a `%inet0` suffix. This has to be replaced by the corresponding `%epairXb` suffix. For now `X = 0` is a good guess. In another environment it could be `X > 0`, of course.

Reply:
I think it might make more sense to replace it the following way then
1) `inet0` -> `epair0b`
2) `inet1` -> `epair1b`